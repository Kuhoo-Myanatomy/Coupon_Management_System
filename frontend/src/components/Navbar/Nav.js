import React,{Component} from "react";
import {Link, withRouter} from 'react-router-dom';

import './nav.css'
class Nav extends Component{
    constructor(props) {
        super(props);
        this.state = {
          currentTab: 'home',
        };
    }
    
    componentDidMount() {
        this.handleNavigation(this.props.location);
    }
    
    componentDidUpdate(prevProps) {
        if (this.props.location !== prevProps.location) {
          this.handleNavigation(this.props.location);
        }
    }
    
    handleNavigation(location) {
        const path = location.pathname;
        if (path === '/') {
          this.setState({ currentTab: 'home' });
        } else if (path === '/coupons') {
          this.setState({ currentTab: 'coupons' });
        } else if (path === '/add') {
          this.setState({ currentTab: 'new' });
        }else if (path === '/login') {
            this.setState({ currentTab: 'login' });
          } else if (path === '/signup') {
            this.setState({ currentTab: 'signup' });
          }else if (path === '/logout') {
            this.setState({ currentTab: 'logout' });
          }
    }
    render(){
        console.log(this.props.location);
        const { currentTab } = this.state;
        console.log(currentTab);
        return(
            <div className="navbar">
                <ul>
                    <li></li>
                    <li> <Link to="/" className={currentTab === 'home' ? 'active-tab' : ''}>Home</Link></li>
                    <li> <Link to="/coupons" className={currentTab === 'coupons' ? 'active-tab' : ''}>Coupons</Link></li>
                    <li> <Link to="/add" className={currentTab === 'new' ? 'active-tab' : ''}>New</Link></li>
                </ul>
                <ul>

                    <li><Link to="/logout" className={currentTab === 'logout' ? 'active-tab' : ''}>Logout</Link></li>
                    <li><Link to="/login" className={currentTab === 'login' ? 'active-tab' : ''}>Login</Link></li>
                    <li><Link to="/signup" className={currentTab === 'signup' ? 'active-tab' : ''}>SignUp</Link></li>
      
                </ul>
            </div>
        )
    }
}

export default withRouter(Nav);