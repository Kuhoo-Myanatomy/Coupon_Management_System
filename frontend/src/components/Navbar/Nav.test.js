import Nav from './Nav';
import{shallow,configure} from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';

configure({ adapter: new Adapter() });

describe("Testing Navbar Component",()=>{
    it("Matching Navbar's snapshot",()=>{
        const component=shallow(<Nav/>);
        expect(component).toMatchSnapshot();
    });
});