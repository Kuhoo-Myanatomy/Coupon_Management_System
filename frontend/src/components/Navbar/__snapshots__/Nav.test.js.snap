// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component Matching Navbar's snapshot 1`] = `ShallowWrapper {}`;
